### PR TITLE
Jetpack Boost: Image Guide compatibility improvements

### PR DESCRIPTION
**Pull request by pyronaur**
Introduce portals to break the Popup out of its containers as various plugins affect the Popup differently.

#### Changes proposed in this Pull Request:

* Add portals
* Use a fixed position for the guide

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Jetpack product discussion
pc9hqz-1sC-p2#comment-1104

#### Does this pull request change what data or activity we track or use?
No

### Todo

- [x] Add external link icon pc9hqz-1sC-p2#comment-1095
- [x] Remove the hideous hack in `initialize.ts`
- [x] Address any other feedback in the thread: pc9hqz-1sC-p2

#### Testing instructions:

Install the following plugins and populate them with content:
* Various Gutenberg galleries
* Gutenberg full-width container with a fixed position background image:

This is a typical page that I use for most testing (apart from the plugins listed below):

https://user-images.githubusercontent.com/988095/208727490-6c5c352a-d7a8-47f9-b59c-1839170002fd.mp4

* Jetpack (Jetpack mosaic/tiled gallery, Jetpack image comparison)
* Smart Slider
* Visual Portfolio with Masonry layout:
CleanShot 2022-12-20 at 19 16 48@2x

### CODE REVIEW

File: projects/plugins/boost/app/features/image-guide/src/initialize.ts [hunk 1]
```diff
@@ -31,8 +30,22 @@ function getClosestContainingAncestor( node: HTMLElement ): HTMLElement | null {
...
+			// The idea here was to reset elementsWithoutZIndex array
+			// whenever we encounter an element that has a z-index set.
+			// But that breaks the portfolio in the portfolio plugin layout.
+			// By not resetting the array, we're keeping
+			// tracks of first static element in the tree.
+			// This seems to work out ok for now,
+			// But at this point, the logic here now
+			// has changed from "reset the array when we encounter a z-indexed element"
+			// to "reset when we encounter a non-static element that's absolutely positioned"
+			// elementsWithoutZIndex = [];
```
Comment (pyronaur) [inline projects/plugins/boost/app/features/image-guide/src/initialize.ts]: `@TODO`

**Review by thingalon — DISMISSED** [review]
I'm ok with these changes, but I think more experimentation is needed. I'm still getting issues in tiled image galleries, and I'm just not sure about the z-index ancestor search logic. I might try to fiddle more with it this week :)

**Review by thingalon — DISMISSED** [review]
😮 this is working, and it's working _well_. It works nicely on Tiled Galleries, and I am all for merging this asap.

Just 🌈 glorious!

I merged from trunk and got some weird monorepo ... glitching. Happy to help untangle that, but I've also raised it with crew here: p1673589594309319-slack-CBG1CP4EN

File: docs/testing/regression-checklist/mapping.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
 		"testFile": [ "extensions/blocks/checklist.md" ]
 	},
 	{
-		"regex": "^projects\/plugins\/jetpack\/extensions\/(blocks\/(subscriptions|premium-content)\/|shared\/memberships\.js)",
+		"regex": "^projects/plugins/jetpack/extensions/(blocks/(subscriptions|premium-content)/|shared/memberships.js)",
```
Comment (dilirity) [inline docs/testing/regression-checklist/mapping.json:33]: The version in trunk is different.
https://github.com/Automattic/jetpack/blob/21c9c9d586c08e25af92f5de3d8122234f4fb4ff/docs/testing/regression-checklist/mapping.json#L33

File: projects/plugins/boost/app/features/image-guide/src/initialize.ts (hunk 1 above)
Comment (dilirity) [reply projects/plugins/boost/app/features/image-guide/src/initialize.ts]: This has been done in d1f6125 and 9103e82.

File: docs/testing/regression-checklist/mapping.json (hunk 2 above)
Comment (dilirity) [reply docs/testing/regression-checklist/mapping.json:33]: @pyronaur which version is correct? The other regexes in the file aren't escaped.

Comment (pyronaur) [reply docs/testing/regression-checklist/mapping.json:33]: I've got no idea what this is and how it got here.

Comment (pyronaur) [reply docs/testing/regression-checklist/mapping.json:33]: We probably need to revert this

Comment (pyronaur) [reply docs/testing/regression-checklist/mapping.json:33]: Ah, I see what you mean, yeah - the others aren't escaped, so i think this'll be fine?

Comment (dilirity) [reply docs/testing/regression-checklist/mapping.json:33]: Prettier formatting did this. Closing.

**Review by dilirity — APPROVED** [review]
Let's merge this.

**Review by dereksmart — APPROVED** [review]
Approving on behalf of the required review for the docs/ update

**Pull request merged**